nengo/base.py

```diff
@@ -54,14 +56,35 @@ class NengoObject(with_metaclass(NetworkMember, SupportDefaultsMixin)):
     seed = IntParam('seed', default=None, optional=True)
 
     def __init__(self, label, seed):
+        super(NengoObject, self).__init__()
         self.label = label
         self.seed = seed
 
     def __getstate__(self):
-        raise NotImplementedError("Nengo objects do not support pickling")
+        state = super(NengoObject, self).__getstate__()
+        del state['_initialized']
+        return state
 
     def __setstate__(self, state):
-        raise NotImplementedError("Nengo objects do not support pickling")
+        from nengo.network import Network
+        super(NengoObject, self).__setstate__(state)
+        setattr(self, '_initialized', True)
+        if len(Network.context) > 0:
+            warnings.warn(
+                "{obj} was not added to the network. When copying objects, "
+                "use the copy method on the object instead of Python's copy "
+                "module. When unpickling objects, they have to be added to "
+                "networks manually.".format(obj=self),
+                NotAddedToNetworkWarning)
```

Comment:
I think the formulation might be ambiguous. If you pickle and unpickle (or copy) a whole network, you should get back the complete network. What is not respected is the current context in which you are unpickling or copying. If you use the `copy` method on the objet or network, it will be respected (and you have the usual `add_to_container` to change this behaviour). The reason that this is not supported by Python's `copy` function:
* There is no way to pass an `add_to_container` argument.
* Not adding the object to the active network allows to use Python's `copy` in the object's `copy` function for implementing both behaviours. Otherwise we would implement all of `copy` ourselves.